src/App.tsx

```diff
@@ -9,56 +9,96 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App: React.FC = () => {
-  return (
-    <main className="section">
-      <div className="container">
-        <div className="tile is-ancestor">
-          <div className="tile is-parent">
-            <div className="tile is-child box is-success">
-              <div className="block">
-                <UserSelector />
-              </div>
+import { CommentsProvider } from './components/Context';
 
-              <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+import { Post } from './types/Post';
+import { User } from './types/User';
 
-                <Loader />
+export const App: React.FC = () => {
+  const [postList, setPostList] = useState<Post[] | undefined>(undefined);
+  const [userSelect, setUserSelect] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [openForm, setOpenForm] = useState(false);
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+  const [hasPostListError, setHasPostListError] = useState(false);
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
+  return (
+    <CommentsProvider>
+      <main className="section">
+        <div className="container">
+          <div className="tile is-ancestor">
+            <div className="tile is-parent">
+              <div className="tile is-child box is-success">
+                <div className="block">
+                  <UserSelector
+                    setPostList={setPostList}
+                    setUserSelect={setUserSelect}
+                    userSelect={userSelect}
+                    setPostListError={setHasPostListError}
+                    setOpenForm={setOpenForm}
+                  />
+                </div>
+                <div className="block" data-cy="MainContent">
```

Comment:
```suggestion
                </div>
                
                <div className="block" data-cy="MainContent">
```
pay attention to the code style